### PR TITLE
Update InVEST to build against numpy 2

### LINKED ISSUE #1641: Update InVEST for numpy2
Now that pygeoprocessing 2.4.5 is released, we should update InVEST to handle numpy 2.

### PR DESCRIPTION
**Pull request by phargogh**
This PR updates InVEST to build against numpy 2.

Fixes #1641

## Checklist
- [x] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
~~- [ ] Updated the user's guide (if needed)~~
~~- [ ] Tested the Workbench UI (if relevant)~~

### CODE REVIEW

**Comment by phargogh** [comment]
There appears to be some overlap here with #1631  ... guess I need to be more careful with my branches!

File: Makefile [hunk 1]
```diff
@@ -199,15 +199,15 @@ $(GIT_SAMPLE_DATA_REPO_PATH): | $(DATA_DIR)
...
+	git -C $(GIT_SAMPLE_DATA_REPO_PATH) lfs checkout $(GIT_SAMPLE_DATA_REPO_REV)
 
 $(GIT_TEST_DATA_REPO_PATH): | $(DATA_DIR)
 	-git clone $(GIT_TEST_DATA_REPO) $(GIT_TEST_DATA_REPO_PATH)
 	git -C $(GIT_TEST_DATA_REPO_PATH) fetch
 	git -C $(GIT_TEST_DATA_REPO_PATH) lfs install
 	git -C $(GIT_TEST_DATA_REPO_PATH) checkout $(GIT_TEST_DATA_REPO_REV)
 	git -C $(GIT_TEST_DATA_REPO_PATH) lfs fetch
-	git -C $(GIT_TEST_DATA_REPO_PATH) lfs checkout
+	git -C $(GIT_TEST_DATA_REPO_PATH) lfs checkout $(GIT_TEST_DATA_REPO_REV)
```
Comment (dcdenu4) [inline Makefile:210]: Is that last `checkout` step needed? We've done the `lfs fetch` and should already be in `$(GIT_TEST_DATA_REPO_REV)` right?

File: pyproject.toml [hunk 2]
```diff
@@ -46,7 +46,8 @@ invest = "natcap.invest.cli:main"
 # available at runtime.
 requires = [
     'setuptools>=61', 'wheel', 'setuptools_scm>=8.0', 'cython>=3.0.0', 'babel',
-    'oldest-supported-numpy'
+    'oldest-supported-numpy; python_version<="3.8"',
+    'numpy>=2; python_version>"3.8"',  # numpy 2 only available for 3.9+
```
Comment (dcdenu4) [inline pyproject.toml:50]: Awesome, and `>3.8` wouldn't grab `3.8.6` or something I'm assuming.

File: src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py [hunk 3]
```diff
@@ -1238,7 +1238,7 @@ def cn_op(lulc_array, soil_group_array):
         # if lulc_array value not in lulc_to_soil[soil_group_id]['lulc_values']
         # then numpy.digitize will not bin properly and cause an IndexError
         # during the reshaping call
-        lulc_unique = set(numpy.unique(lulc_array))
+        lulc_unique = set(i.item() for i in numpy.unique(lulc_array))
```
Comment (dcdenu4) [inline src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py:1241]: I tried to look this up quickly, but I'm not sure I follow calling `item()` here...

Comment (dcdenu4) [reply src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py:1241]: Ah, unless we're concerned with datatypes, I bet that's it.

File: src/natcap/invest/urban_flood_risk_mitigation.py [hunk 4]
```diff
@@ -936,7 +936,8 @@ def validate(args, limit_to=None):
         nan_mask = cn_df.isna()
         if nan_mask.any(axis=None):
             nan_lucodes = nan_mask[nan_mask.any(axis=1)].index
-            lucode_list = list(nan_lucodes.values)
+            # Convert numpy dtype values to native python types
```
Comment (dcdenu4) [inline src/natcap/invest/urban_flood_risk_mitigation.py:939]: Yep!

**Review by dcdenu4 — COMMENTED** [review]
Thanks @phargogh ! I had one small question around sample / test data in the Makefile.

File: Makefile (hunk 1 above)
Comment (phargogh) [reply Makefile:210]: Yep, it should be fine without this change!  Removed in 89d236b1f

File: src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py (hunk 3 above)
Comment (phargogh) [reply src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py:1241]: Yep, that's it exactly!  `i` would normally be a native numpy object (e.g. `numpy.float32`), so calling `.item()` gets the python representation of the object.

**Review by dcdenu4 — APPROVED** [review]
This looks good! Thanks!

**Pull request merged**